packages/tenant-process/src/services/tenantService.ts

```diff
@@ -510,6 +514,100 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async revokeVerifiedAttribute(
+      {
+        tenantId,
+        attributeId,
+        organizationId,
+        correlationId,
+      }: {
+        tenantId: TenantId;
+        attributeId: AttributeId;
+        organizationId: TenantId;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<Tenant> {
+      logger.info(
+        `Revoke verified attribute ${attributeId} to tenant ${tenantId}`
+      );
+
+      if (organizationId === tenantId) {
+        throw verifiedAttributeSelfRevocation();
+      }
+
+      const allowedStatuses = [
+        agreementState.pending,
+        agreementState.active,
+        agreementState.suspended,
+      ];
+
+      await assertVerifiedAttributeOperationAllowed({
+        producerId: organizationId,
+        consumerId: tenantId,
+        attributeId,
+        agreementStates: allowedStatuses,
+        readModelService,
+        error: attributeRevocationNotAllowed(tenantId, attributeId),
+      });
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      const verifiedTenantAttribute = targetTenant.data.attributes.find(
+        (attr): attr is VerifiedTenantAttribute =>
+          attr.type === tenantAttributeType.VERIFIED && attr.id === attributeId
+      );
+
+      if (!verifiedTenantAttribute) {
+        throw attributeNotFound(attributeId);
+      }
+
+      const verifier = verifiedTenantAttribute.verifiedBy.find(
+        (a) => a.id === organizationId
+      );
+
+      if (!verifier) {
+        throw attributeRevocationNotAllowed(tenantId, attributeId);
+      }
+
+      if (
+        verifiedTenantAttribute.revokedBy.find((a) => a.id === organizationId)
+      ) {
+        throw attributeAlreadyRevoked(tenantId, organizationId, attributeId);
+      }
+
+      const updatedTenant = {
+        ...targetTenant.data,
+        createdAt: new Date(),
+        attributes: targetTenant.data.attributes.map((attr) =>
+          attr.id === attributeId
+            ? ({
+                ...verifiedTenantAttribute,
+                verifiedBy: verifiedTenantAttribute.verifiedBy.filter(
+                  (v) => v.id !== organizationId
+                ),
+                revokedBy: [
+                  ...verifiedTenantAttribute.revokedBy,
+                  {
+                    ...verifier,
+                    revocationDate: new Date(),
+                  },
+                ],
+              } as VerifiedTenantAttribute)
+            : attr
```

Comment:
Let's avoid using `as` at all costs.
Using `as` here means we lose the typesafetiness from line 584 to line 596, this could lead to bug which are difficult to track. 
Example, you here could make a typo in the `verifiedBy` property, typescript would not complain. 

Possible solution, using `satisfies` instead of `as`.